### PR TITLE
Implement dml delete parsing

### PR DESCRIPTION
**Pull request by if-loop69420**
## Summary
Add support for dml ```DELETE``` statements and DML in general
## What has been done
- all the things in #118 , but more foundational stuff like adding dml in general 
---

### CODE REVIEW

File: src/grammar/dml.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#[allow(unused)]
+pub(crate) fn parse_dml(p: &mut Parser) {
+    parse_delete(p);
+}
+
+pub(crate) fn parse_delete(p: &mut Parser) {
+    p.start(SyntaxKind::DeleteStmt);
+    p.expect(T![delete]);
+    p.expect(T![from]);
+    p.expect_one_of(&[T![quoted_ident], T![unquoted_ident]]);
```
Comment (KieranKaelin) [inline src/grammar/dml.rs:16]: The table may be schema qualified, such as `hr.emp`. For this we have the `parse_ident` helper function.

File: src/grammar/dml.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+use source_gen::lexer::TokenKind;
+use source_gen::syntax::SyntaxKind;
+use source_gen::T;
+
+#[allow(unused)]
+pub(crate) fn parse_dml(p: &mut Parser) {
+    parse_delete(p);
+}
+
+pub(crate) fn parse_delete(p: &mut Parser) {
```
Comment (KieranKaelin) [inline src/grammar/dml.rs:12]: Please add this to `parse_stmt`, which in turn is used by `parse_block`, including tests.
`parse_block` is used by all kinds of constructs, such as procedures and packages.

**Review by KieranKaelin — CHANGES_REQUESTED** [review]

File: src/ast/dml.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    pub fn where_clause(&self) -> Option<WhereClause> {
+        self.syntax.children().find_map(WhereClause::cast)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::ast::{AstNode, Root};
+
+    // use super::*;
```
Comment (KieranKaelin) [inline src/ast/dml.rs:16]:
```suggestion
```

**Review by KieranKaelin — CHANGES_REQUESTED** [review]

File: src/grammar/dml.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
+use super::{parse_ident, parse_where_clause};
+use crate::parser::Parser;
+use source_gen::lexer::TokenKind;
+use source_gen::syntax::SyntaxKind;
+use source_gen::T;
+
+#[allow(unused)]
+pub(crate) fn parse_dml(p: &mut Parser) {
+    parse_delete(p);
+}
```
[the remark this answers is not in the document]
Comment (if-loop69420) [reply src/grammar/dml.rs:10]: This is a placeholder. We add parse_update as an alternate branch in the other pr

[the remark this answers is not in the document]
Comment (if-loop69420) [reply src/grammar/dml.rs:10]: #118

File: src/grammar/dml.rs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+pub(crate) fn parse_delete(p: &mut Parser) {
+    p.start(SyntaxKind::DeleteStmt);
+    p.expect(T![delete]);
+    p.expect(T![from]);
+    parse_ident(p, 1..1);
```
[the remark this answers is not in the document]
Comment (if-loop69420) [reply src/grammar/dml.rs:16]: wdym?

[the remark this answers is not in the document]
Comment (if-loop69420) [reply src/grammar/dml.rs:16]: so parse ident then check for "." and parse_ident again?

[the remark this answers is not in the document]
Comment (KieranKaelin) [reply src/grammar/dml.rs:16]: Check the function docs and/or usages, you can describe your expected identifier using the range parameter.

**Review by KieranKaelin — APPROVED** [review]

**Pull request merged**